src/pathfinding.h

```diff
@@ -60,12 +61,15 @@ struct pathfinding_settings {
     bool avoid_traps = false;
     bool allow_climb_stairs = true;
     bool avoid_rough_terrain = false;
+    bool avoid_sharp = false;
 
     pathfinding_settings() = default;
     pathfinding_settings( const pathfinding_settings & ) = default;
-    pathfinding_settings( int bs, int md, int ml, int cc, bool aod, bool at, bool acs, bool art )
+    pathfinding_settings( int bs, int md, int ml, int cc, bool aod, bool at, bool acs, bool art,
+                          bool as )
```

Comment:
The fifth boolean argument is a little bit too much here (actually, even a single one is usually evil). It's impossible to figure the meaning of the train of `true`s and `false`s without looking at the declaration. I'd suggest either dropping all the constructors in favor of explicit assignments:

```c++
pathfinding_settings ps;
ps.allow_open_doors = true;
ps.avoid_traps = false;
...
```
or using a bitset of named flags for this purpose.

Reply:
thanks, I'll open an issue for this, this probably could be done for a number of methods like this.